inference-engine/thirdparty/clDNN/src/gpu/ocl_builder.cpp

```diff
@@ -89,9 +101,20 @@ std::map<std::string, device_impl::ptr>  ocl_builder::build_device_list_from_use
 
     std::map<std::string, device_impl::ptr> ret;
     uint32_t idx = 0;
+    uint32_t last_dev_idx = (uint32_t)all_devices.size(), di = 0;
     for (auto& device : all_devices) {
         if (!does_device_match_config(out_out_order, device)) continue;
-        ret.insert(get_device(idx++, device, device.getInfo<CL_DEVICE_PLATFORM>()));
+        auto flag = device.getInfo<CL_DEVICE_HOST_UNIFIED_MEMORY>();
+        if (flag != 0)
+            last_dev_idx = di;
+        else
+            ret.insert(get_device(idx++, device, device.getInfo<CL_DEVICE_PLATFORM>()));
+        di++;
```

Comment:
`di` will increment incorrectly if one of devices does not meet the requirements because of this statement `if (!does_device_match_config(out_out_order, device)) continue; ` in the loop

Reply:
Thanks, fixed.